### PR TITLE
[amazon] Add Amazon.templefile

### PR DESCRIPTION
**Pull request by jaylees14**
Added my first stab at the Amazon templefile - a few things missing:
- `enumerable(by: )` would be good
- Limiting write access to certain services to specific kind of auth 

Couple of other things that popped into my head
- We don't yet support multiple services with auth - but this is a classic case where we would want to
- It was quite hard to break down into multiple services with join tables. I'm wondering if we could introduce a List<Product> type and we generate the join table? Just a thought

### CODE REVIEW

File: amazon.temple [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Amazon: project {
+    #language(go);
+    #database(postgres);
+}
+
+Seller: service {
+    firstName: string;
+    lastName: string;
+
+    #auth(email);
```
Comment (Lewiky) [inline amazon.temple]: is the email field implicit? I thought you still had to put it into the service block explicitly

File: amazon.temple [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Order: service {
+    customer: Customer;
+    orderDate: datetime;
+
+    #omit([delete]);
+}
+
+// OrderItem is a join table between Orders and Products (many to many)
+OrderItem: service {
```
Comment (xsanda) [inline amazon.temple]: Is it right for this to be a service in its own right? Microservices are weird

File: amazon.temple [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    addressLine1: string;
+    addressLine2: string;
+    town: string;
+    county: string;
+    postcode: string;
+
+    #auth(email);
+}
+
+ProductCategory: service {
```
Comment (xsanda) [inline amazon.temple]: Should this be a struct?

File: amazon.temple (hunk 1 above)
Comment (jaylees14) [reply amazon.temple]: The email field is indeed implicit - it gets thrown into another service called `Auth`

**Pull request merged**